doc/enterprise/auth/auth.md

```diff
@@ -0,0 +1,234 @@
+# Understanding Pachyderm access controls
+
+If access controls are activated, each data repository, or repo,
+in Pachyderm has an Access Control List (ACL) associated with it.
+The ACL includes:
+
+- `READERs` - users who can read the data versioned in the repo.
+- `WRITERs` - users with `READER` access who can also submit
+additions, deletions, or modifications of data into the repo.
+- `OWNERs` - users with READER and WRITER access who can also
+modify the repo's ACL.
+
+Pachyderm defines the following user account types:
+
+* **GitHub user** is a user account that is associated with
+a GitHub account and logs in through the GitHub OAuth flow. If you do not
+use any third-party identity provider, you use this option. When a user tries
+to log in with a GitHub account, the system verifies the identity and
+sends a Pachyderm token for that account.
+* **Robot user** is a user account that logs by using a pach-generated authentication
+token. Typically, you create a user in simplified workflow scenarios, such
+as initial SAML configuration.
+* **Pipeline** is a user account that Pachyderm creates for your
```

Comment:
nit: I know this 100% contradicts the language that I used in my comment, but now that I see it written out, I think it might make sense to refer to pipelines as pipelines as an "account" type, but not a "user account" type (there's no way for a user to say "I am pipeline:foo", unlike e.g. robot accounts, where a user can get a token from an admin and say "I am robot:foo")

Reply:
Agree. Removed the word `user`.